src/mixins/java/org/spongepowered/common/mixin/core/entity/player/ServerPlayerEntityMixin.java

```diff
@@ -468,26 +495,144 @@ public void teleportTo(net.minecraft.world.server.ServerWorld world, double x, d
+                this.shadow$unRide();
+                this.shadow$getLevel().removePlayerImmediately((ServerPlayerEntity) (Object) this);
+                if (!this.wonGame) {
+                    this.wonGame = true;
+                    this.connection.send(new SChangeGameStatePacket(SChangeGameStatePacket.WIN_GAME, this.seenCredits ? 0.0F : 1.0F));
+                    this.seenCredits = true;
+                }
+
+                // TODO: Figure out the reposition step (though that might just be a respawn)
+                return (Entity) (Object) this;
+            } else {
+                IWorldInfo iworldinfo = targetWorld.getLevelData();
+                this.connection.send(new SRespawnPacket(targetWorld.dimensionType(), targetWorld.dimension(),
+                        BiomeManager.obfuscateSeed(targetWorld.getSeed()), this.gameMode.getGameModeForPlayer(),
+                        this.gameMode.getPreviousGameModeForPlayer(), targetWorld.isDebug(), targetWorld.isFlat(), true));
+                this.connection.send(new SServerDifficultyPacket(iworldinfo.getDifficulty(), iworldinfo.isDifficultyLocked()));
+                PlayerList playerlist = this.server.getPlayerList();
+                playerlist.sendPlayerPermissionLevel((ServerPlayerEntity) (Object) this);
+                serverworld.removePlayerImmediately((ServerPlayerEntity) (Object) this);
+                this.removed = false;
+                // Sponge Start: Redirect the find portal call to the teleporter.
+
+                // TODO: This could house the Reposition Event, with some flag on SpongeTeleporter that can be set to cause this to happen later
+                //  for mods.
+                //  This can then hold our custom portal logic (basically any edits to findDimensionEntryPoint that we might need to do)
+                final PortalInfo portalinfo = spongeTeleporter.getPortalInfo((Entity) (Object) this, targetWorld, this::shadow$findDimensionEntryPoint);
+                // Sponge End
+                if (portalinfo != null) {
+                    // Sponge Start: wrap the teleportation logic within a function to allow for modification
+                    // of the teleporter, as well as calling events(?)
+                    final Vector3d originalDestination = new Vector3d(portalinfo.pos.x, portalinfo.pos.y, portalinfo.pos.z);
+                    final Entity e = spongeTeleporter.performTeleport((Entity) (Object) this, serverworld, targetWorld, this.xRot, spawnPortal -> {
+                        // Sponge End
+                        serverworld.getProfiler().push("moving");
+                        if (registrykey == World.OVERWORLD && targetWorld.dimension() == World.NETHER) {
+                            this.enteredNetherPosition = this.shadow$position();
+                        // Sponge Start: only enter this branch if we're going through portal logic
+                        } else if (spawnPortal && targetWorld.dimension() == World.END) {
```

Comment:
I heavily debated running this for Vanilla's END or END dimension types (since they will use the END generator).

The problem is the Vanilla code, no matter what, puts the platform there. Someone could get angry in a custom end having a build at  spawn get overwrote due to this. For Vanilla END, we can state that is game mechanics.